packages/p2p/src/components/file-uploader-component/__tests__/file-uploader-component.spec.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { StoreProvider, mockStore } from '@deriv/stores';
+import FileUploaderComponent from '../file-uploader-component';
+
+describe('<FileUploaderComponent />', () => {
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    const file = new File(['hello'], 'hello.png', { type: 'image/png' });
+    const props = {
+        accept: 'image/pdf, image/png',
+        filename_limit: 26,
+        hover_message: 'drop here',
+        max_size: 2097152,
+        multiple: false,
+        setDocumentFile: jest.fn(),
+        validation_error_message: null,
+        upload_message: 'upload here',
+        value: [],
+    };
+
+    it('should render FileUploaderComponent component in desktop mode', async () => {
+        render(<FileUploaderComponent {...props} />, {
+            wrapper: ({ children }) => <StoreProvider store={mockStore({})}>{children}</StoreProvider>,
+        });
+        expect(screen.getByText('upload here')).toBeInTheDocument();
+    });
+
+    it('should upload supported file', async () => {
+        const new_props = {
+            ...props,
+            value: [file],
+        };
+        render(<FileUploaderComponent {...new_props} />, {
+            wrapper: ({ children }) => <StoreProvider store={mockStore({})}>{children}</StoreProvider>,
+        });
+        const input: HTMLInputElement = screen.getByTestId('dt_file_upload_input');
```

Comment:
Can we use getbyrole query here? it seems like it's an input

Reply:
checked this. the role is not accessible.